pkg/provision/storage/testdata/perWorkspace-storage/use-default-size-when-calculated-size-smaller-than-default.yaml

```diff
@@ -0,0 +1,70 @@
+name: "Uses default PVC size when not all volumes define their size"
+
+input:
+  devworkspaceId: "test-workspaceid"
+  podAdditions:
+    containers:
+      - name: testing-container-1
+        image: testing-image
+        volumeMounts:
+          - name: "projects"
+            mountPath: "/projects-mountpath"
+          - name: "volume-1"
+            mountPath: "/test-1"
+    initContainers:
+      - name: testing-initContainer-1
+        image: testing-image
+        volumeMounts:
+          - name: "plugins"
+            mountPath: "/plugins"
+          - name: "volume-1"
+            mountPath: "/test-3"
+
+  workspace:
+    components:
+      - name: testing-container-1
+        container:
+          image: testing-image-1
+          sourceMapping: "/plugins-mountpath"
+      - name: volume-1
+        volume:
+          size: 1Gi
+      - name: volume-2
+        volume:
+          size: 1Gi
+      - name: volume-3
+        volume:
+          size: 512Mi
+      - name: volume-4
+        volume:
+          size: 248Mi
+      - name: plugins
+        volume: {}
+      # Calculated PVC size is 2760Mi, which is smaller than default 5Gi, so default should be used
```

Comment:
Two issues here:

1. The output does not define a size, so the computed size is not checked for this test (we only check the returned PVC size if it's not nil)
2. If the devfile requests less storage than the configured default, we _do_ want to use the smaller amount. This would allow, for example, attaching a small amount of storage to a web terminal.

I think this test file should be dropped, as the code as written currently behaves according to point 2 above and would fail this test.

Reply:
>     1. The output does not define a size, so the computed size is not checked for this test (we only check the returned PVC size if it's not nil)

Unless I'm mistaken, I added [a check](https://github.com/devfile/devworkspace-operator/pull/1020/files#diff-6074799d5832e0da6614320e8779bcacb5c0a169069bee9820670c970f36b636R91) so that if the output does not define a PVC size _and_ we're not expecting an error, we check that the retrieved PVC size is the same as the default PVC size. 

>     2. If the devfile requests less storage than the configured default, we _do_ want to use the smaller amount. This would allow, for example, attaching a small amount of storage to a web terminal.

I'll make the appropriate change, as I forgot that you had [mentioned that](https://github.com/devfile/devworkspace-operator/issues/947#issuecomment-1380570915). Currently, you only can use a smaller amount if all volumes have their size defined (ignoring ephemeral volumes). 

> I think this test file should be dropped, as the code as written currently behaves according to point 2 above and would fail this test.

:+1:

Reply:
@amisevsk to be clear: if not all volume sizes are specified (but at least one volume size is) and the computed PVC size is smaller than the default, we want to use the computed PVC size?

Reply:
Ahh I see, I misread the test. Nevermind!